cql/src/main/java/org/apache/cassandra/stargate/metrics/ClientMetrics.java

```diff
@@ -183,39 +220,79 @@ private int countConnectedClients() {
+   *
+   * @param name metric name
+   * @return full name
+   */
+  private static String metric(String name) {
+    // to keep the back-ward compatibility, init all metric names with cql_ + DefaultNameFactory
+    String metricName = factory.createMetricName(name).getMetricName();
+    return "cql." + metricName;
   }
 
-  private List<Map<String, String>> recentClientStats() {
-    List<Map<String, String>> stats = new ArrayList<>();
+  private class ConnectionMetricsImpl implements ConnectionMetrics {
+
+    private final Counter requestsProcessed;
+    private final Counter requestsDiscarded;
+    private final Counter authSuccess;
+    private final Counter authFailure;
+    private final Counter authError;
+    private final AtomicInteger connectedClients;
+
+    public ConnectionMetricsImpl(ClientInfo clientInfo) {
+      Tags tags =
+          Optional.ofNullable(clientInfo)
+              .map(clientInfoTagProvider::getClientInfoTags)
+              .orElse(Tags.empty());
+
+      requestsProcessed = meterRegistry.counter(REQUESTS_PROCESSED_METRIC, tags);
+      requestsDiscarded = meterRegistry.counter(REQUESTS_DISCARDED_METRIC, tags);
+      authSuccess = meterRegistry.counter(AUTH_SUCCESS_METRIC, tags);
+      authFailure = meterRegistry.counter(AUTH_FAILURE_METRIC, tags);
+      authError = meterRegistry.counter(AUTH_ERROR_METRIC, tags);
+      connectedClients =
+          meterRegistry.gauge(CONNECTED_NATIVE_CLIENTS_METRIC, tags, new AtomicInteger(0));
+    }
 
-    for (Server server : servers)
-      for (ClientStat stat : server.recentClientStats()) stats.add(stat.asMap());
+    @Override
+    public void markRequestProcessed() {
+      requestsProcessed.increment();
+    }
 
-    stats.sort(Comparator.comparing(map -> map.get(ClientStat.PROTOCOL_VERSION)));
+    @Override
+    public void markRequestDiscarded() {
+      requestsDiscarded.increment();
+    }
 
-    return stats;
-  }
+    @Override
+    public void markAuthSuccess() {
+      authSuccess.increment();
+    }
 
-  private <T> Gauge<T> registerGauge(String name, Gauge<T> gauge) {
-    return metricRegistry.register(factory.createMetricName(name).getMetricName(), gauge);
-  }
+    @Override
+    public void markAuthFailure() {
+      authFailure.increment();
+    }
 
-  private Histogram registerHistogram(String name) {
-    return metricRegistry.histogram(factory.createMetricName(name).getMetricName());
-  }
+    @Override
+    public void markAuthError() {
+      authError.increment();
+    }
 
-  private Counter registerCounter(String name) {
-    return metricRegistry.counter(factory.createMetricName(name).getMetricName());
-  }
+    @Override
+    public void increaseConnectedNativeClients() {
+      connectedClients.incrementAndGet();
+    }
 
-  private Meter registerMeter(String name) {
-    return metricRegistry.meter(factory.createMetricName(name).getMetricName());
+    @Override
+    public void decreaseConnectedNativeClients() {
```

Comment:
Is this method called? I could not find the call side, sorry.

Reply:
Removed.